### PR TITLE
corr_map function docstring v0.1

### LINKED ISSUE #8: Correlation maps - docstring
correlation maps 
input: features to be correlated; 
output: correlation maps with Altair/matplot

### PR DESCRIPTION
**Pull request by ChuckHo777**

### CODE REVIEW

File: simpler_eda/corr_map.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import pandas as pd
+import altair as alt
+import numpy as np
+
+
+def corr_map(path, data_type, features, sep=",", color_theme='yellowgreenblue', plot_width=400, plot_height=300, title="Correlation Map"):
```
Comment (nichowu) [inline simpler_eda/corr_map.py]: 1. removing "reading dataset"
2. color_theme : str, optional
3. "Plot a correlation map for the given list of features. Users can also set multiple arguments regarding the data (e.g. the separator) and the setting for the correlation plot from colour schemes, plot width, height and plot title.

File: simpler_eda/corr_map.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    """
+    Plot a correlation map for the given list of feature and raw (the path and data type). 
+    Users can also set multiple arugment regarding the data (e.g. the separator) and the setting for the correlation plot
+    from color schemes, plot withh & height, title.
+
+    Parameters
+    ----------
+    path: str
```
Comment (nichowu) [inline simpler_eda/corr_map.py]: remove path

File: simpler_eda/corr_map.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    data: str
+      The data type for the data. 
+      It only supports CSV and Json at the moment.
+
+    features: list
+      A 1D list of numerical feature names in string with the feature names for correlation map plotting.
+      It should contain at least 2 features.
+
+    color_theme: str, default='yellowgreenblue'
```
Comment (nichowu) [inline simpler_eda/corr_map.py]: color_scheme

File: simpler_eda/corr_map.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      The color scheme
+      Other color schemes can be "blues", "tealblues", "oranges", "greenblue", "redpurple", etc.
+      Other proper color scheme reference can be found inhttps://vega.github.io/vega/docs/schemes/
```
Comment (nichowu) [inline simpler_eda/corr_map.py]:
```suggestion
      Other proper colour scheme reference can be found in https://vega.github.io/vega/docs/schemes/
```

File: simpler_eda/corr_map.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    Returns
+    -------
+    altair.vegalite.v4.api.Chart
+      The altair correlation map plot
+
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> import altair as alt
+    >>> from corr_map import corr_map
```
Comment (nichowu) [inline simpler_eda/corr_map.py]: from simpler_eda.corr_map import corr_map

File: simpler_eda/corr_map.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+    Returns
+    -------
+    altair.vegalite.v4.api.Chart
+      The altair correlation map plot
+
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> import altair as alt
```
Comment (nichowu) [inline simpler_eda/corr_map.py]: add:
import numpy as np

File: simpler_eda/corr_map.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    >>> from corr_map import corr_map
+    >>> corr_map.corr_map("data/abc.csv", "csv", ["age", "height", "income"])
```
Comment (nichowu) [inline simpler_eda/corr_map.py]: remove "corr_map."

File: simpler_eda/corr_map.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    An altair.vegalite.v4.api.Chart object
```
Comment (nichowu) [inline simpler_eda/corr_map.py]: Returns: chart: Chart object altair.vegalite.v4.api.Chart
                The altair Chart object build from the specification

File: simpler_eda/corr_map.py (hunk 4 above)
Comment (ChuckHo777) [reply simpler_eda/corr_map.py]: As discussed, I will edit all comments in one go and PR again while color should be kept as color.

File: simpler_eda/corr_map.py (hunk 8 above)
Comment (ChuckHo777) [reply simpler_eda/corr_map.py]: As discussed, let's keep the example ending with calling the function.

File: simpler_eda/corr_map.py [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+  plot_height: int, optional
+    The heigh of the plot
+
+  title: str, optional
+    The title of the correlation map
+
+  Returns
+  -------
+  altair.vegalite.v4.api.Chart
```
Comment (nichowu) [inline simpler_eda/corr_map.py]: change to:

Returns
--------
`altair`
The altair correlation map plot

File: simpler_eda/corr_map.py [hunk 10]
```diff
@@ -0,0 +1,51 @@
+import pandas as pd
+import altair as alt
+import numpy as np
+
+
+def corr_map(data, features, color_schme='yellowgreenblue', plot_width=400, plot_height=300, title="Correlation Map"):
+  """
+  Plot a correlation map with the given dataframe object and a list of numerical features. 
+  Users are allow to set multiple arguments regarding the setting of the correlation plot
```
Comment (nichowu) [inline simpler_eda/corr_map.py]: Users are **_allowed_** to set multiple arguments regarding the setting of the correlation plot **_including_** color schemes, plot width, height and plot title.

**Review by nichowu — CHANGES_REQUESTED** [review]
2 changes requested.

**Review by nichowu — APPROVED** [review]

**Pull request merged**